### ISSUE TITLE
package build with -v master param fails

### ISSUE THREAD

**Issue by augustf** [concerto/concerto-debian#62]
In the Debian build script(https://github.com/concerto/concerto-debian/blob/master/build_deb_packages.sh#L10), it's mentioned that the -v option can be passed master, but this does not seem to work. @mfrederickson, Any memory of how we had this doing that?

```
./build_deb_packages.sh -v master

Building packages for VERSION master but repo is at 2.3.3 !!

  initializing and updating submodules...
Submodule 'concerto_full/usr/share/concerto' (git://github.com/concerto/concerto.git) registered for path 'concerto_full/usr/share/concerto'
Submodule 'concerto_lite/usr/share/concerto' (git://github.com/concerto/concerto.git) registered for path 'concerto_lite/usr/share/concerto'
Cloning into 'concerto_full/usr/share/concerto'...
remote: Counting objects: 30057, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 30057 (delta 0), reused 0 (delta 0), pack-reused 30054
Receiving objects: 100% (30057/30057), 12.42 MiB | 5.90 MiB/s, done.
Resolving deltas: 100% (17026/17026), done.
Checking connectivity... done.
Submodule path 'concerto_full/usr/share/concerto': checked out '0dc71dfa3a5d20cfd67854c740fcb9a45b7f0063'
Cloning into 'concerto_lite/usr/share/concerto'...
remote: Counting objects: 30057, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 30057 (delta 0), reused 0 (delta 0), pack-reused 30054
Receiving objects: 100% (30057/30057), 12.42 MiB | 5.76 MiB/s, done.
Resolving deltas: 100% (17026/17026), done.
Checking connectivity... done.
Submodule path 'concerto_lite/usr/share/concerto': checked out '0dc71dfa3a5d20cfd67854c740fcb9a45b7f0063'

Building concerto-full package...

  updating concerto_full's local repo...
Already up-to-date.
  local repo at 2.3.3-1-gd5c419e
  copying service script from repo...
  setting permissions for control files...
  building package...
dpkg-deb: error: parsing file 'concerto_full/DEBIAN/control' near line 2 package 'concerto-full':
 error in 'Version' field string 'master': version number does not start with digit
  checking package... (results logged to: full_lintian.log)
warning: cannot find binary, udeb or source package debs/concerto-full_master_all.deb in lab (skipping)
    0 errors
    0 warnings
    0 overrides

Building concerto-lite package...

  updating concerto_lite's local repo...
Already up-to-date.
  local repo at 2.3.3-1-gd5c419e
  copying service script from repo...
  setting permissions for control files...
  building package...
dpkg-deb: error: parsing file 'concerto_lite/DEBIAN/control' near line 2 package 'concerto-lite':
 error in 'Version' field string 'master': version number does not start with digit
  checking package... (results logged to: lite_lintian.log)
warning: cannot find binary, udeb or source package debs/concerto-lite_master_all.deb in lab (skipping)
    0 errors
    0 warnings
    0 overrides

Preparing Packages for Deployment...

  removing old packaging...
  preparing concerto_full package...
Error 2 opening ../debs/concerto-full_master_all.deb: No such file or directory
There have been errors!
  preparing concerto_lite package...
Error 2 opening ../debs/concerto-lite_master_all.deb: No such file or directory
There have been errors!

finished
```

**Comment by mfrederickson** [comment]
I seem to recall 0.0.0 or something like that being used when specifying master... I'll dredge around some more...

**Comment by mfrederickson** [comment]
August-- see the [2014] commit comment I just made and mentioned you on...

**Comment by mfrederickson** [comment]
@augustf It looks like need to revert this to make it work https://github.com/concerto/concerto-debian/commit/e38d4ed517cf6341978bc995feb15004c8cc42e1

**Comment by augustf** [comment]
@mfrederickson That line I removed didn't make any sense - we don't want our versions prepended by all those zeroes. Perhaps once I tag a version this will work. I'll have to test...

**Comment by mfrederickson** [comment]
One of the packaging tools requires a number for the version in one of the control files.  Version tags work fine-- because it gets the latest.  But when you use a branch name-- even master-- it needs some type of number for the packaging tool.  How do we know what number to use?  Should that be another parameter that is passed in?

**Comment by mfrederickson** [comment]
https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version

>The version number of a package. The format is: [epoch:]upstream_version[-debian_revision]
...
>upstream_version
>This is the main part of the version number. It is usually the version number of the original ("upstream") package from which the .deb file has been made, if this is applicable. Usually this will be in the same format as that specified by the upstream author(s); however, it may need to be reformatted to fit into the package management system's format and comparison scheme.
>
>The comparison behavior of the package management system with respect to the upstream_version is described below. The upstream_version portion of the version number is mandatory.
>
>**The upstream_version may contain only alphanumerics[36] and the characters . + - : ~** (full stop, plus, hyphen, colon, tilde) and should start with a digit. If there is no debian_revision then hyphens are not allowed; if there is no epoch then colons are not allowed.
>...
>It is conventional to restart the debian_revision at 1 each time the upstream_version is increased.
>...
>Note that the purpose of epochs is to allow us to leave behind mistakes in version numbering, and to cope with situations where the version numbering scheme changes. It is not intended to cope with **version numbers containing strings of letters which the package management system cannot interpret (such as ALPHA or pre-),** or with silly orderings.[38]

**Comment by augustf** [comment]
Hmm-I'll have to take a closer look at this. Whatever is assigned will need to be in keeping with our existing versioning nomenclature. Aside from the obvious sorts of things in the version that the packager won't accept, we don't (and really couldn't) follow the Debian guidelines on versioning since we're not under their governance.

**Comment by mfrederickson** [comment]
Running `./build_deb_package.sh` without a parameter will build the latest released version into a deb.  Running it with `-v master` will build the master branch into a deb (and the version in the deb will be 0.0.0 so it wont work with testing an upgrade-- you'd have to change it to something higher than the current version).  And if you pass it `-v x.y.z` where x.y.z is some semantic version label of an existing version in the concerto github repo, then it will build that version into a deb.

**Comment by mfrederickson** [comment]
Closed with #75
